add linearity tests for calibration data

TellAnAx opened the pull request:
Guten Tag Herr Ranke,
wie besprochen einmal ein PR, zunächst einmal für einen Linearitätstest. Ich dachte wir können erstmal klein anfangen, um das Ganze überschaubar zu halten. 
Ich bin gespannt auf das Feedback!
Mit besten Grüssen,
A. Tellbüscher

TellAnAx committed:
add linearity tests for calibration data

jranke committed:
Use match.arg for checking method argument

jranke commented:
I have only briefly looked at the code and have taken the liberty of modify the PR by using the match.arg() function to check the `method` argument because I think it will be easier for others to understand what the code does.

As far as the tests are concerned, I am surprised that the point to point slope for the data set used in the example shows so many points outside the tolerance range. From my point of view, this does not mean that the method is not linear, it simply has a high scatter. What do you think?

jranke committed:
Merge branch 'main' into pull_request

jranke committed:
Merge branch 'main' into pull_request

TellAnAx commented:
Great, I was not aware of the `match.arg()` function. This is a much more elegant way and thus much appreciated! 

Regarding the test results: Yes, you are right that the `din32645` data is not particularly well-suited to showcase the use of the `linearity()` function. This could be better done by using one of the example datasets provided in the Excel sheet for linearity testing written by the Water Chemistry Society ([here](https://www.wasserchemische-gesellschaft.de/images/PDFs/DEV/Validierungsdokumente%20und%20Arbeitshilfen/A51_DIN_38402_51_2017_Linearitaetstest_v2.1.xlsx)). The tolerance range of 10% also depends on the analytical method chosen. Unfortunately, I do not have access to a copy of DIN 32 645. Do you know the method that was used to acquire the data? Regarding the function itself: Would a more detailed explanation do the job in this case? What do you think?

jranke commented:
Adding a comment to the example code would be an option. Another possibility would be to write a small vignette on linearity tests. Including another dataset also could be useful. However, as I said, I cannot invest a lot of time in this package, there would be so many things that could be done to improve it, but I am not doing analytics any more.

jranke commented:
Hi this is easy, just push to the branch you used to create the PR.

Am Montag, 25. November 2024, 13:22:34 CET schrieb TellAnAx:
> Alright! My time is also limited, but, as mentioned, I am willing to
> contribute more to this package. I don't know whether the analytic
> community waited for this, but at least I was looking for a good toolbox to
> analyse my data for way too long now, so better doing the job myself
> instead of waiting forever! :) I will add some additional example data
> then. I just need to figure out how to add an additional commit to this
> pull request! :)

TellAnAx committed:
add .Rproj file

TellAnAx committed:
import graphics::lines

TellAnAx committed:
add .Rproj file

TellAnAx committed:
add example calibration data from DIN 38402

TellAnAx committed:
replace din32645 data by data from din38402b1

jranke committed:
Satisfy R CMD check

- Depend on R >= 3.5.0, because the format used for the data is not
  known to earlier R versions
- Rename the datasets from `test` to their proper name

jranke committed:
Add Anil A. Tellbuescher as author, NEWS

jranke committed:
Use author name in DESCRIPTION as spelled by himself
